### PR TITLE
Add checkov static analysis

### PR DESCRIPTION
**Pull request by kwong0222**
- First pass at adding checkov to tf workflow
- Report on failure
- Use ENV var instead of outputs
- Set output as var
- Use env variable instead of local
- trigger new workflow
- Add input to skip specified check_ids
- Add comment if checkov passes
- double quote var
- Set output as var before evaulating
- double quotes
- add documentation

Example output can be seen in this test PR https://github.com/rewindio/terraform-sourcegraph/pull/38

### CODE REVIEW

File: .github/workflows/checkov.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
+name: IaC static analysis 
+
+on:
+  workflow_call:
+    inputs:
+      skipped_check_ids:
+        description: "Comma separated Checkov check IDs to skip"
+        required: false
+        type: string
```
Comment (davegallant) [inline .github/workflows/checkov.yml:9]: To keep consistency between local machines and CI, would it be worth advocating for a [.checkov.yml](https://github.com/bridgecrewio/checkov#configuration-using-a-config-file)?

Comment (kwong0222) [reply .github/workflows/checkov.yml:9]: I like the idea! A config would keep the CI .yml file clean and , as you mentioned, consistency across environments. I'll remove this input and have the action read a config file instead.

Do you find a comment with in the PR with which checks were skipped helpful? I guess it wont be needed if those check ids were defined in the yml file.

**Review by davegallant — APPROVED** [review]

File: .github/workflows/checkov.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+          ref: ${{ steps.load-pr.outputs.head_sha }}
+          token: ${{ secrets.GITHUB_PAT }} 
+      
+      - name: Setup Git Config
+        run: |
+          git config --global \
+            url."https://oauth2:${{ secrets.GITHUB_PAT }}@github.com".insteadOf https://github.com
```
Comment (davegallant) [inline .github/workflows/checkov.yml:27]: I don't think this is needed.

Comment (davegallant) [reply .github/workflows/checkov.yml:27]: or does checkov pull private terraform modules?

Comment (kwong0222) [reply .github/workflows/checkov.yml:27]: There is a config for external modules `download-external-modules`. In the docs, it says that flag is used to download *public* repos.

When I run checkov locally, the `terraform-rewindio-globals` gets pulled. My assumption was that if git was configured with a token, it should be able to pull private modules.

Comment (davegallant) [reply .github/workflows/checkov.yml:27]: Ooh then this would be necessary then.

**Pull request merged**